jolie/src/jolie/process/ReThrowProcess.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2016 Martin Wolf.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301  USA
+ */
+package jolie.process;
+
+import jolie.ExecutionThread;
+import jolie.lang.Constants;
+import jolie.runtime.ExitingException;
+import jolie.runtime.FaultException;
+import jolie.runtime.Value;
+import jolie.runtime.VariablePathBuilder;
+import jolie.runtime.expression.Expression;
+
+public class ReThrowProcess implements Process
+{
+	private final Expression expression;
+
+	public ReThrowProcess( Expression expression )
+	{
+		this.expression = expression;
+	}
+
+	public ReThrowProcess()
+	{
+		this.expression = null;
+	}
+
+	@Override
+	public void run()
+		throws FaultException, ExitingException
+	{
+		if ( ExecutionThread.currentThread().isKilled() ) {
+			return;
+		}
+
+		ExecutionThread ethread = ExecutionThread.currentThread();
+		if ( expression == null ) {
+			throw ethread.currentFault();
```

Comment:
What if currentFault() is null?

Which reminds me: we should probably insert a check in the semantic verifier that rethrow is used only from inside of something that is being installed using the `install` statement.